cartridges/int_algolia_sfra/cartridge/static/default/js/algolia/recommend-config.js

```diff
@@ -0,0 +1,383 @@
+        type: 'trendingItems',
+        containerId: 'trendingItems',
+        recommendClient: config.recommendClient,
+        facetName: '__primary_category.' + categoryDepth,
+        facetValue: categoryHierarchy
+    });
+    createRecommendationWidget({
+        type: 'similarContents',
+        containerId: 'similarContents',
+        recommendClient: config.recommendClient
+    });
+    createRecommendationWidget({
+        type: 'relatedProducts',
+        containerId: 'relatedProducts',
+        recommendClient: config.recommendClient
+    });
+    createRecommendationWidget({
+        type: 'lookingSimilar',
+        containerId: 'lookingSimilar',
+        recommendClient: config.recommendClient
+    });
+}
+
+/**
+ * Generic function to create a recommendation widget
+ * @param {Object} options - Options for the widget creation
+ * @returns {void}
+ */
+function createRecommendationWidget(options) {
+    const { type, containerId, recommendClient, facetName, facetValue } = options;
+    const container = document.getElementById(containerId);
+    if (!container) return;
+
+    const indexName = algoliaData.productsIndex;
+    const objectIDs = getObjectIds(container);
+
+    if (type === 'frequentlyBoughtTogether' && !objectIDs) return;
+
+    if (type === 'relatedProducts' && !objectIDs) return;
+
+    if (type === 'lookingSimilar' && !objectIDs) return;
+
+    const widgetOptions = {
+        container: `#${containerId}`,
+        recommendClient,
+        indexName,
+        objectIDs,
+        maxRecommendations: 4,
+        itemComponent: ({ item, html }) => itemComponent({ item, html })
+    };
+
+    if (type === 'frequentlyBoughtTogether') {
+        frequentlyBoughtTogether(widgetOptions);
+    } else if (type === 'trendingItems') {
+        if (facetValue && facetValue !== '' && facetValue !== ' > ') {
+            widgetOptions.facetName = facetName;
+            widgetOptions.facetValue = facetValue;
+        }
+        trendingItems(widgetOptions);
+    } else if (type === 'similarContents') {
+        widgetOptions.indexName = algoliaData.contentsIndex;
+        widgetOptions.itemComponent = ({ item, html }) => contentComponent({ item, html });
+        relatedProducts(widgetOptions);
+        const observerConfig = {
+            containerId,
+            type: 'childList',
+            observationTarget: '.auc-Recommend-title',
+            observationText: algoliaData.strings.relatedProducts,
+            observationReplacement: algoliaData.strings.relatedContent
+        }
+        widgetObserver(observerConfig);
+    } else if (type === 'relatedProducts') {
+        relatedProducts(widgetOptions);
+    } else if (type === 'lookingSimilar') {
+        lookingSimilar(widgetOptions);
+    }
+}
+
+/**
+ * Widget observer for modifiying the widgets default appearance
+ * @param {Object} observerConfig - Configuration object
+ * @returns {void}
+ */
+function widgetObserver(observerConfig) {
```

Comment:
Can you explain in more detail (here and in the comment) what is this observer used for? I see that it changes the title of the widget, but in which cases can that happen? I'd like to test it.

Reply:
Good catch! Its purpose was modifying header title for similar content but I just noticed that I can also use `headerComponent`. Code is refactored and this function is removed